docs_outdoornav_user_manual/faq.mdx

```diff
@@ -33,52 +33,161 @@ toc_max_heading_level: 4
+    1.2 m/s, 1.0 m/s and 2.0 m/s for Jackal, Husky and Warthog UGVs, respectively. The Husky maximum velocity
+    is 1.0 m/s so increasing above this is not possible however if you would like to increase the maximum
+    velocity for either the Jackal or the Warthog, further tuning may be required. Consult the 
+    [tuning guide](./customized_tuning/tuning_instructions/differential_drive_dynamics) and contact
+    [customer support](./support) if required.
 
-7.  **The UGV position and/or heading are behaving erratically while the
-    UGV is stationary. What should I do?**
+4.  **My robot it detecting too many obstacles and replanning too frequently?**
 
-    Power cycle the UGV by powering off the base platform. Wait 10
-    seconds, then turn the base power back on, wait 10 seconds, then
-    turn the computer power back on (labelled **PWR** on the port/right
-    side of mast). Allow 1-2 minutes for the software to startup and
-    refresh the Web UI page.
+    Please consult the [Limitations](./features/collision_avoidance#limitations) section of the collisison
+    avoidance feature. It will describe some limits to the obstaacle detection and can help you improve the 
+    smoothness of the navigation.
 
-## Autonomous Docking
+5.  **I am getting a 'Robot too far off path warning'. What should I do?**
 
-1.  **When trying to dock, my UGV drives towards the dock but stops 2
-    meters away and remains there. "Docking Failed" is displayed in
-    the UI feedback bar. How can I fix this?**
+    These types of warnings appears on the UI under two conditions:
+    i)  If the robot indicator (blue arrow on the UI) is not within 3.0 meters of the first Waypoint (the Green one on the UI).
+    ii) If the robot is outside of the allowable navigable space (defined by the path contraint, defualt: 3.0 meters around the reference path). 
+    Enable the visualization of the [path contraint](./web_user_interface/ui_autonomous_mode#constrained_path) Teleop the robot back into the the 
 
-    It is likely that one of these 3 things occurred:
+6.  **How can I remove certain sensors from the collision avoidance pipeline?**
 
-    1.  The Base Station was recently resurveyed, or
-    2.  The datum was recently changed, or
-    3.  The dock location was not set correctly.
+    If your robot is equipped with collisions avoidance sensors (eg. Velodyne, Hokuyo, Sick LMS), it is
+    possible to enable/disable the throughput of each sensor data into the collision avoidance pipeline.
+    Consult the [Collision Avoidance](./features/collision_avoidance) section for these instructions.
+
+7.  **Can I use OudoorNav without using the UI?**
+
+    We empower customers to send mission via either the UI or our [ROS1 API](./api/api_overview). Through this
+    API, you are enabled to write Python scripts where you would generate your mission, survey the base station,

```

Comment:
C++ and Python? (Noting that our example code is only Python.)

Reply:
are we able to use CPP to import the api_libraries that you wrote? I was not aware that we could do this.

I guess they could write their own CPP files that do everything manually but I don't think they would be able to use your libraries

Reply:
Agreed that they can't use my api_libraries with CPP, but if they really want to write CPP they can do that and the ROS API should work. They just wouldn't have the benefit of api_libraries.